### PR TITLE
[tracegen/cmd] support add additional resource attributes.

### LINKED ISSUE #11144: [tracegen/cmd] support add additional resource attributes in tracegen
**Is your feature request related to a problem? Please describe.**
I noticed that [tracegen](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/tracegen) cmd does not support adding custom resource attributes, which I think is more important in helping users develop and debug.

**Describe the solution you'd like**
I'd love to work/add this support in `tracegen` cmd.  
I think we can refer the implementation of `otlp-headers` and maybe like this:
```bash
tracegen -otlp-insecure -duration 5s -otlp-attributes key1=value1 -otlp-attributes key2=value2
```

### PR DESCRIPTION
**Pull request by JaredTan95**
Signed-off-by: jian.tan <jian.tan@daocloud.io>

**Description:** <Describe what has changed.>

refer the implementation of `otlp-headers` :
```bash
tracegen -otlp-insecure -duration 5s -otlp-attributes key1=value1 -otlp-attributes key2=value2
```

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11144

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]
LGTM!

File: tracegen/internal/tracegen/config.go [hunk 1]
```diff
@@ -78,6 +79,11 @@ func (c *Config) Flags(fs *flag.FlagSet) {
 	c.Headers = make(map[string]string)
 	fs.Var(&c.Headers, "otlp-header", "Custom header to be passed along with each OTLP request. The value is expected in the format key=value."+
 		"Flag may be repeated to set multiple headers (e.g -otlp-header key1=value1 -otlp-header key2=value2)")
+
+	// custom resource attributes
+	c.ResourceAttributes = make(map[string]string)
```
Comment (mx-psi) [inline tracegen/internal/tracegen/config.go:84]: Resource attributes can have other types like e.g. integers or booleans. I think it's fine to only add support for strings first, but we should think about how we will add support for other types in the future. 

To make it concrete: right now `-otlp-attributes key=true` will create an attribute with the string value `"true"`, which can be confusing. Should we force people to add quotes to wrap the values? (so .e.g. you have to do `key="true"` for adding a string vs `key=true` for adding a literal boolean?)

Comment (mx-psi) [reply tracegen/internal/tracegen/config.go:84]: Actually maybe this gets solved if you do `map[string]interface{}` instead of `map[string]string`?

Comment (mx-psi) [reply tracegen/internal/tracegen/config.go:84]: It's fine to not support these now, but we need to be able to detect them now. My suggestion is to do the following:

```
func (v *KeyValue) Set(s string) error {
	kv := strings.SplitN(s, "=", 2)
	if len(kv) != 2 {
		return fmt.Errorf("value should be of the format key=\"value\"")
	}
        val := kv[1]
        if len(val) < 2 || val[0] != "\"" || val[len(val) - 1] != "\" {
                return fmt.Errorf("value should be a string wrapped in double quotes")
        }

	(*v)[kv[0]] = val[val[1 : len(val)-1]]
	return nil
}
```

In this way, `key="true"` will pass (setting to the string value `"true"`) while `key=true` will fail (and we can support it later).

Comment (JaredTan95) [reply tracegen/internal/tracegen/config.go:89]: Thanks for your review, after adding this logic:
```bash
-otlp-insecure -duration 5s -otlp-attributes key1=\"value2\" -otlp-attributes key2=\"123\" -otlp-attributes key3=\"true\" 
```

File: tracegen/internal/tracegen/config.go [hunk 2]
```diff
@@ -78,6 +84,11 @@ func (c *Config) Flags(fs *flag.FlagSet) {
...
+	fs.Var(&c.ResourceAttributes, "otlp-attributes", "Custom resource attributes to use. The value is expected in the format key=value."+
+		"Flag may be repeated to set multiple attributes (e.g -otlp-attributes key1=value1 -otlp-attributes key2=value2)")
```
Comment (mx-psi) [inline tracegen/internal/tracegen/config.go:91]:
```suggestion
	fs.Var(&c.ResourceAttributes, "otlp-attributes", "Custom resource attributes to use. The value is expected in the format key=\"value\"."+
		"Flag may be repeated to set multiple attributes (e.g -otlp-attributes key1=\"value1\" -otlp-attributes key2=\"value2\")")
```

File: tracegen/main.go [hunk 3]
```diff
@@ -100,8 +102,18 @@ func main() {
 		}
 	}()
 
+	var attributes []attribute.KeyValue
+	// may be overridden by `-otlp-attributes service.name=foo`
```
Comment (mx-psi) [inline tracegen/main.go:106]:
```suggestion
	// may be overridden by `-otlp-attributes service.name="foo"`
```

**Review by mx-psi — COMMENTED** [review]
Looks good, just a few doc updates after the quotes thing

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
Thanks @JaredTan95 for contributing! :)

**Pull request merged**